wm/desk.go

```diff
@@ -431,8 +450,8 @@ func (x *x11WM) handleStateActionRequest(ev xproto.ClientMessageEvent, removeSta
 func (x *x11WM) handleInitialHints(ev xproto.ClientMessageEvent, hint string) {
 	switch clientMessageStateAction(ev.Data.Data32[0]) {
 	case clientMessageStateActionRemove:
-		hints := x.pendingRemoveHints[ev.Window]
-		hints = append(hints, hint)
+		windowExtendedHintsRemove(x.x, ev.Window, hint)
+		x.showWindow(ev.Window)
```

Comment:
yes - showWindow is required because the window is hidden and _NET_WM_HIDDEN remove is telling us to show the window... so showWindow will setup a client by calling setupWindow and setupWindow calls for a new frame which shows the window.